tools/plugins/mgm/src/main/kotlin/net/corda/cli/plugins/mgm/BaseOnboard.kt

```diff
@@ -0,0 +1,284 @@
+    }
+
+    private val url by lazy {
+        val rpcPort = if (networkName != null) {
+            val port = ServerSocket(0).use {
+                it.localPort
+            }
+            ProcessBuilder().command(
+                "kubectl",
+                "port-forward",
+                "--namespace",
+                networkName,
+                "deployment/corda-rpc-worker",
+                "$port:8888"
+            )
+                .inheritIO()
+                .start().also { process ->
+                    Runtime.getRuntime().addShutdownHook(
+                        thread(false) {
+                            process.destroy()
+                        }
+                    )
+                }
+            Thread.sleep(2000)
+            port
+        } else {
+            8888
+        }
+
+        "https://localhost:$rpcPort/api/v1"
+    }
+
+    protected fun setupClient() {
+        Unirest.config()
+            .verifySsl(false)
+            .setDefaultBasicAuth("admin", "admin")
+            .defaultBaseUrl(url)
+    }
+
+    internal class OnboardException(message: String) : Exception(message)
+
+    internal fun <T> HttpResponse<T>.bodyOrThrow(): T {
+        if (!this.isSuccess) {
+            throw OnboardException("Error: ${this.body}")
+        }
+        return this.body
+    }
+    protected fun uploadCpi(cpi: InputStream, name: String): String {
+        val id = cpi.use { jarInputStream ->
+
+            Unirest.post("/cpi")
+                .field("upload", jarInputStream, "$name.cpb")
+                .asJson()
+                .bodyOrThrow()
+                .`object`.get("id")
+        }
+        val end = System.currentTimeMillis() + 5 * 60 * 1000
+        while (System.currentTimeMillis() < end) {
+            val response = Unirest.get("/cpi/status/$id").asJson()
+            val status = try {
+                response.bodyOrThrow().`object`.get("status")
+            } catch (e: OnboardException) {
+                "Not ready yet"
+            }
+            if (status == "OK") {
+                return response.body.`object`.get("cpiFileChecksum").toString()
+            }
+        }
+
+        throw OnboardException("CPI request $id had failed!")
+    }
+
+    protected abstract val cpiFileChecksum: String
+
+    abstract var x500Name: String
+
+    protected abstract val registrationContext: Map<String, Any?>
+
+    protected val holdingId by lazy {
+        val body = mapOf(
+            "request" to mapOf(
+                "cpiFileChecksum" to cpiFileChecksum,
+                "x500Name" to x500Name
+            )
+        ).let {
+            json.writeValueAsString(it)
+        }
+        Unirest.post("/virtualnode")
+            .body(body)
+            .asJson()
+            .bodyOrThrow()
+            .let {
+                it.`object`.let { reply ->
+                    if (reply.get("state") != "ACTIVE") {
+                        throw OnboardException("MGM node is not active")
```

Comment:
MGM mentioned in common code
```suggestion
                        throw OnboardException("Virtual node is not active")
```